rune/libenclave/intelsgx/aesmd.go

```diff
@@ -135,6 +135,59 @@ func DumpLocalReport(report []byte) error {
 	return nil
 }
 
+func DumpQuote(quote []byte) error {
+	q := &Quote{}
+	if err := restruct.Unpack(quote, binary.LittleEndian, &q); err != nil {
+		return err
+	}
+
+	logrus.Debugf("QUOTE:")
+	logrus.Debugf("  Version:                              %d\n",
+		q.Version)
+	logrus.Debugf("  Attestation Key Type:                 %d\n",
+		q.AttestationKeyType)
+
+	if q.Version == QuoteVersion2 {
+		quoteBody := &QuoteBodyV2{}
+		if err := restruct.Unpack(quote[QuoteHeaderLength:QuoteHeaderLength+QuoteBodyLength], binary.LittleEndian, &quoteBody); err != nil {
+			return err
+		}
+
+		logrus.Debugf("  Gid:                                  %#08x\n",
+			quoteBody.Gid)
+		logrus.Debugf("  ISV assigned SVN for Quoting Enclave: %d\n",
+			quoteBody.ISVSvnQe)
+		logrus.Debugf("  ISV assigned SVN for PCE:             %d\n",
+			quoteBody.ISVSvnPce)
+		logrus.Debugf("  Base name:                            0x%v\n",
+			hex.EncodeToString(quoteBody.Basename[:]))
+	} else if q.Version == QuoteVersion3 {
+		quoteBody := &QuoteBodyV3{}
+		if err := restruct.Unpack(quote[QuoteHeaderLength:QuoteHeaderLength+QuoteBodyLength], binary.LittleEndian, &quoteBody); err != nil {
+			return err
+		}
+
+		logrus.Debugf("  Quoting Enclave SVN:                  %d\n",
+			quoteBody.QeSvn)
+		logrus.Debugf("  PCE SVN:                              %d\n",
+			quoteBody.PceSvn)
+		logrus.Debugf("  Quoting Enclave Vendor Id:            0x%v\n",
+			hex.EncodeToString(quoteBody.QeVendorId[:]))
+		logrus.Debugf("  User Data:                            0x%v\n",
+			hex.EncodeToString(quoteBody.UserData[:]))
+
+	} else {
+		return fmt.Errorf("Unsupported Quote Version: %d", q.Version)
+	}
+
+	logrus.Debugf("  Report:                               ...\n")
```

Comment:
If you decide to dump the report, plz remove this line.

Reply:
However, the dump messages for local report have a wrong indent. So dump function should have a parameter for indent size to get a good result view.

Reply:
> However, the dump messages for local report have a wrong indent. So dump function should have a parameter for indent size to get a good result view.

This issue is not solved now, I will add it as soon as possible @jiazhang0 .